Installation process fix

webpatser opened the pull request:
Just did a clean install of v1.0. And this step is in the wrong order. 

First enable the filament panel. Then do the `lunar:install`. Otherwise an error occurs with the lunar:install, stating the panel is needed. But then it's unclear that you need to run the installer again to seed the database with needed permissions and users etc.

webpatser committed:
- incorrect order of steps. This way no error in `lunar:install`

webpatser committed:
- removed some PHPstorm auto-formating line wrap `helpers` changes

webpatser committed:
- added admin creation in `lunar:install` if no admin exists.
- Fixed exits in sub commands, stopping the whole install command, without any warning.

webpatser commented:
Since this PR was pending. I upgraded the whole install process. It now has correct return types in the sub-commands.
Also a check for a admin user was added, and if no admin is present, create one.

webpatser committed:
- typo

webpatser committed:
Merge branch '1.x' into installation_docs

webpatser committed:
Merge branch '1.x' into installation_docs

alecritson committed:
Merge branch '1.x' into installation_docs

webpatser committed:
- missed failure exits in two commands

webpatser committed:
Merge branch '1.x' into installation_docs

eugenefvdm commented:
Thanks so much for this.

I notice when running this pull request that `DB::transaction` still fails because of missing attributes, specifically `description`:

```mysql
SQLSTATE[HY000]: General error: 1364 Field 'description' doesn't have a default value (Connection: mysql, SQL: insert into `lunar_attributes` (`attribute_type`, `attribute_group_id`, `position`, `name`, `handle`, `section`, `type`, `required`, `default_value`, `configuration`, `system`, `updated_at`, `created_at`) values (Lunar\Models\Product, 1, 1, {"en":"Name"}, name, main, Lunar\FieldTypes\TranslatedText, 1, ?, {"richtext":false}, 1, 2024-02-11 06:24:53, 2024-02-11 06:24:53))
```

On my system the temporary fix for all attributes was to append a `description` field such as the example below:

```php
 Attribute::create([
            'attribute_type' => Product::class,
            'attribute_group_id' => $group->id,
            'position' => 1,
            'name' => [
                'en' => 'Name',
            ],
            'handle' => 'name',
            'section' => 'main',
            'type' => TranslatedText::class,
            'required' => true,
            'default_value' => null,
            'configuration' => [
                'richtext' => false,
            ],
            'system' => true,
            'description' => '', // TODO had to add this to make the seeder work
        ]);
```

I had to do this for all four attributes seeded in the installer to make it work.

From what I can see in the [1.x documentation](https://docs-v1.lunarphp.io/core/reference/attributes.html) description is maybe only partially documented. In the `MySQL` it's a JSON field that cannot be null.

I would be grateful to know what we can use as initial values for this `description`.

eugenefvdm commented:
Something else to consider to make a working install are tax defaults, apart from the `TaxClass`.

This is present in the Livewire starter kit:
https://github.com/lunarphp/livewire-starter-kit/blob/main/database/seeders/TaxSeeder.php

webpatser commented:
I will add the description. I didn't had that problem. But lots of merges last week, so maybe something changed upstream.

glennjacobs commented:
Thanks @webpatser for the help. We'll be working on this soon (this coming week) and this is a helpful start.

webpatser committed:
- smal refactor to follow Laravel Command setup

webpatser committed:
- smal refactor to follow Laravel Command setup

webpatser committed:
Merge branch '1.x' into installation_docs

webpatser committed:
- adding newly added Attributes `description` field to the installer

webpatser commented:
Added the new `description` field to the Lunar installer. That field was recently added in `2024_01_11_100000_add_description_to_attributes_table.php`.

For now the description is blank, this should maybe contain some default description :-)

glennjacobs commented on packages/core/src/Console/Commands/MakeLunarAdminCommand.php:
```diff
@@ -0,0 +1,91 @@
+use function Laravel\Prompts\password;
+use function Laravel\Prompts\text;
+
+class MakeLunarAdminCommand extends Command
```
This command should be in the admin package, as it references the Staff model.

glennjacobs commented on packages/core/src/Console/InstallLunar.php:
```diff
@@ -63,6 +64,12 @@ public function handle(): void
 
         DB::transaction(function () {
+
+            if (! Staff::whereAdmin(true)->exists()) {
```
We can't assume the admin package is installed

glennjacobs replied:
Some developers only install the core package.

glennjacobs commented on packages/core/src/Console/Commands/MakeLunarAdminCommand.php:
```diff
@@ -0,0 +1,91 @@
+        $this->options = $this->options();
+
+        if (! Filament::getCurrentPanel()) {
+            $this->error('Filament has not been installed yet: php artisan filament:install --panels');
```
This isn't required. The only thing required for Lunar's panel is to publish the Filament assets, which can be achieved by running

```
php artisan filament:assets
```

glennjacobs commented on packages/core/src/Console/Commands/AddonsDiscover.php:
```diff
@@ -33,5 +33,7 @@ public function handle(Manifest $manifest): void
 
         $this->components->info('Addon manifest generated successfully.');
+
+        return Command::SUCCESS;
```
Laravel now returns void from the handle method, see https://laravel.com/docs/10.x/artisan#command-structure

I suggest we simply omit the return statement now.

webpatser committed:
Merge branch '1.x' into installation_docs

* 1.x:
  Feature - Move Stripe addon into mono repo (#1551)

webpatser committed:
- feedback fixes.
- MakeLunarAdminCommand is now in admin package.
- InstallLunar now checks is Staff model exists.
- InstallLunar now published filament assets in the last step.
- Back to void returns

webpatser commented:
Feedback is processed..

webpatser committed:
Merge branch '1.x' into installation_docs

* 1.x:
  Feature - Discount editing (#1542)
  Update ManageProductShipping.php
  Feature - Prevent delete action on resources with childs (#1532)
  Feature - Brand urls and tweaks (#1541)
  Chore - Update larastan/larastan (#1558)
  chore: fix code style
  Feature - Field translated text (#1505)
  Fix tests
  Revert "Quantity break fixes"
  Quantity break fixes
  Feature - Pricing management (#1515)
  Improved dashboard KPIs

glennjacobs committed:
Fix commands

glennjacobs committed:
Fix tests
